### ISSUE TITLE
No longer defaulting to os.environ

### ISSUE THREAD

**Issue by dkarchmer** [joke2k/django-environ#119]
In v0.4.1, I could do:

```
if os.path.isfile(env_file):
    print('Reading Env file: {0}'.format(env_file))
    environ.Env.read_env(env_file)
else:
    print('Warning!! No .env file: {0}'.format(env_file))
```

and if no environment file existed, it would default to `os.environ`.

This was changed by 6932e1337934c876a6d35f5d735192000fa11623 where line 66 in `environ.py` was changed from:

```
ENVIRON = os.environ
```

to:

```
 ENVIRON = {}
```

If this is not a bug, then this is a major change in behavior, to the point that you should have call this `v0.5.0` and not just `v0.4.2`. Furthermore, you should have documented this as a breaking change (as we wasted a bunch of time tracking this down)

I really think that the old behavior was very useful for testing when it is useful to use OS env variables, especially when testing on the cloud (e.g. even more with Docker based testing like Codeship Pro), and where we don't usually need any real secrets. If there is a reason why doing this by default is wrong, you could at least make this an option in the constructor.

Anyway, current hack (or maybe this is the right way) is to do:

```
env = environ.Env(
   ...
)
env.ENVIRON = os.environ
...
if os.path.isfile(env_file):
    print('Reading Env file: {0}'.format(env_file))
    environ.Env.read_env(env_file)
else:
    print('Warning!! No .env file: {0}'.format(env_file))
```

**Comment by dkarchmer** [comment]
Here is an example of a docker-compose file for testing, where it is useful to keep everything in the same file, while using a real .env for production:

```
version: '2'
services:

  web:
    build:
      context: ./server/
      dockerfile: Dockerfile
    volumes:
      - ./logs:/var/app/logs
      - ./server:/var/app
      - ./staticfiles:/www/static
    ports:
      - "5000:8080"
    environment:
      - DJANGO_SETTINGS_MODULE=config.settings.test
      - DJANGO_SECRET_KEY=test-dummy-key1
      - PRODUCTION=False
      - DEBUG=True
      - DATABASE_DEFAULT_URL=sqlite:///db1.sqlite3
      - DATABASE_STREAMDATA_URL=sqlite:///db2.sqlite3
      - SECRET_KEY=no-key
      - GOOGLE_API_KEY=no-key
      - RECAPTCHA_SITE_KEY=no-key
      - RECAPTCHA_SECRET_KEY=no-key
    command: py.test --cov-config .coveragerc --cov=apps .
```

while the actual `.production.env` has

```
SECRET_KEY=<actual-key>
GOOGLE_API_KEY=<actual-key>
RECAPTCHA_SITE_KEY=<actual-key>
RECAPTCHA_SECRET_KEY=<actual-key>
...
```

Yes, I can just put this on a test.env under version control, but now I have to handle two files, for each testing flavor I may want to use.

**Comment by jbzdak** [comment]
Got bit by it in developer environment also.

**Comment by QasimK** [comment]
#117 (first issue) and #118 related

**Comment by zlove** [comment]
Reading from os.environ by default is useful. In many cloud environments (e.g. Heroku) configuration variables are set in the environment, and there is no need for a separate .env file. At least that is how I do it. 

I'd vote for reverting this change in behavior to not require an .env file, and read from os.environ by default. If this behavior is made explicit and controlled with a flag passed in env = environ.Env(...), that would work for me as well.

@dkarchmer Thanks for the workaround!

**Comment by gegenschall** [comment]
Also got bit by it out of the blue. One would assume that having `django-environ~=0.4.1` is a safe thing to do, but apparently it's not. Mhm. :/

**Comment by edmorley** [comment]
> Also got bit by it out of the blue. One would assume that having django-environ~=0.4.1 is a safe thing to do, but apparently it's not. Mhm. :/

Always pin your dependencies to an exact version, and ideally use pip v8+'s `--require-hashes` mode:
https://pip.pypa.io/en/stable/reference/pip_install/#hash-checking-mode
https://pip.pypa.io/en/stable/user_guide/#repeatability

**Comment by hammadzz** [comment]
Will there be a fix or the hack is the way to go?

**Comment by jamesdh** [comment]
@dkarchmer This was ultimately fixed via https://github.com/joke2k/django-environ/commit/04829316a106583ae559604247a68425c4c19045 and subsequently released as part of v0.4.3

@joke2k it seems this issue can be closed.
